packages/elements-core/src/utils/http-spec/IServer.ts

```diff
@@ -57,3 +61,28 @@ export const getServerUrlWithDefaultValues = (server: IServer): string | null =>
 
   return stringifiedUrl.endsWith('/') ? stringifiedUrl.slice(0, -1) : stringifiedUrl;
 };
+
+export const getServerUrlWithFilledVariables = (server: IServer, values?: { [key: string]: string }): string | null => {
+  let urlString = server.url;
+
+  const variables = Object.entries(server.variables ?? {});
+  variables.forEach(([variableName, variableInfo]) => {
+    urlString = urlString.replace(`{${variableName}}`, values?.[variableName] || variableInfo.default);
```

Comment:
```suggestion
    urlString = urlString.replaceAll(`{${variableName}}`, values?.[variableName] || variableInfo.default);
```
I believe this should be `replaceAll` in case one makes use of variable a number of times.

Reply:
`getServerUrlWithDefaultValues` has the same bug in there too - we could fix it while at it.